Tests/JExtractSwiftTests/FFM/FFMSubscriptsTests.swift

```diff
@@ -0,0 +1,314 @@
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift.org project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import JExtractSwiftLib
+import Testing
+
+import JExtractSwiftLib
+import Testing
+
+@Suite
+struct FFMSubscriptsTests {
+  private let noParamsSubscriptSource = """
+  public struct MyStruct {
+    private var testVariable: Double = 0
+
+    public subscript() -> Double {
+      get { return testVariable }
+      set { testVariable = newValue }
+    }
+  }
+  """
+
+  private let subscriptWithParamsSource = """
+  public struct MyStruct {
+    private var testVariable: [Int32] = []
+
+    public subscript(index: Int32) -> Int32 {
+      get { return testVariable[Int(index)] }
+      set { testVariable[Int(index)] = newValue }
+    }
+  }
+  """
+
+  @Test("Test generation of JavaClass for subscript with no parameters")
+  func generatesJavaClassForNoParams() throws {
+    try assertOutput(
+      input: noParamsSubscriptSource, .ffm, .java,
+      expectedChunks: [
+        """
+        // Generated by jextract-swift
+        // Swift module: SwiftModule
+
+        package com.example.swift;
+
+        import org.swift.swiftkit.core.*;
+        import org.swift.swiftkit.core.util.*;
+        import org.swift.swiftkit.ffm.*;
+        import org.swift.swiftkit.core.annotations.*;
+        import java.lang.foreign.*;
+        import java.lang.invoke.*;
+        import java.util.*;
+        import java.nio.charset.StandardCharsets;
+        """,])
+    try assertOutput(
+      input: noParamsSubscriptSource, .ffm, .java, 
+      expectedChunks: [
+        """
+        public final class MyStruct extends FFMSwiftInstance implements SwiftValue {
+          static final String LIB_NAME = "SwiftModule";
+          static final Arena LIBRARY_ARENA = Arena.ofAuto();
+          @SuppressWarnings("unused")
+          private static final boolean INITIALIZED_LIBS = initializeLibs();
+          static boolean initializeLibs() {
+            System.loadLibrary(SwiftLibraries.LIB_NAME_SWIFT_CORE);
+            System.loadLibrary(SwiftLibraries.LIB_NAME_SWIFT_RUNTIME_FUNCTIONS);
+            System.loadLibrary(LIB_NAME);
+             return true;
+          }
+        """,
+        """
+        public static final SwiftAnyType TYPE_METADATA =
+            new SwiftAnyType(SwiftRuntime.swiftjava.getType("SwiftModule", "MyStruct"));
+        public final SwiftAnyType $swiftType() {
+          return TYPE_METADATA;
+        }
+
+        public static final GroupLayout $LAYOUT = (GroupLayout) SwiftValueWitnessTable.layoutOfSwiftType(TYPE_METADATA.$memorySegment());
+        public final GroupLayout $layout() {
+          return $LAYOUT;
+        }
+
+        private MyStruct(MemorySegment segment, AllocatingSwiftArena arena) {
+          super(segment, arena);
+        }
+        """
+      ])
```

Comment:
Please remove those thunks too, let's not assert anything that's not related to the actual subscripts -- like these chunks